### PR TITLE
Vertex, Matrix classes implemented & graph class changes

### LINKED ISSUE #22: Matix & graph implementation

### PR DESCRIPTION
**Pull request by Secret333Boy**
Issue #16  
Closes #22

### CODE REVIEW

File: src/tools/dataStructures/Matrix.js [hunk 1]
```diff
@@ -0,0 +1,99 @@
...
+    return this.arr;
+  }
+
+  transpone() {
+    const res = [];
+    for (let i = 0; i < this.arr.length; i++) {
+      for (let j = 0; j < this.arr[i].length; j++) {
+        res[j][i] = this.arr[i][j];
+      }
+    }
```
Comment (GrigoriyUstymenko) [inline src/tools/dataStructures/Matrix.js:78]:
```suggestion
    for (const i in this.arr) {
      for (const j in this.arr[i]) {
        res[j][i] = this.arr[i][j];
      }
    }
```

File: src/tools/dataStructures/Matrix.js [hunk 2]
```diff
@@ -0,0 +1,99 @@
...
+      throw new TypeError('Must be an instance of Matrix');
+    }
+
+    if (this.xLength !== matrix.yLength || this.yLength !== matrix.xLength) {
+      throw new Error('Imposible to multiply matrixes with unapropiate size');
+    }
+
+    for (let i = 0; i < this.arr.length; i++) {
+      for (let j = 0; j < this.arr[i].length; j++) {
+        this.arr[i][j] = this.arr[i][0] + matrix.arr[0][j];
```
Comment (GrigoriyUstymenko) [inline src/tools/dataStructures/Matrix.js:62]: Why not multiplication? Also use for in.

**Review by GrigoriyUstymenko — CHANGES_REQUESTED** [review]
All in all good, needs some minor changes.

File: src/tools/dataStructures/Matrix.js [hunk 3]
```diff
@@ -0,0 +1,99 @@
...
+      while (row.length !== maxRowLength) {
+        row.push(null);
+      }
+      res.push(row);
+    }
+    return res;
+  }
+
+  getElement(x, y) {
+    return this.arr[(y, x)];
```
[the remark this answers is not in the document]
Comment (Secret333Boy) [reply src/tools/dataStructures/Matrix.js]: Just a typo. I should test first

File: src/tools/dataStructures/Matrix.js [hunk 4]
```diff
@@ -0,0 +1,99 @@
...
+
+    if (this.xLength !== matrix.xLength || this.yLength !== matrix.yLength) {
+      throw new Error('Imposible to add matrixes with different size');
+    }
+
+    for (let i = 0; i < this.arr.length; i++) {
+      for (let j = 0; j < this.arr[i].length; j++) {
+        this.arr[i][j] += matrix.arr[i][j];
+      }
+    }
```
[the remark this answers is not in the document]
Comment (Secret333Boy) [reply src/tools/dataStructures/Matrix.js:47]: Good idea!

**Review by GrigoriyUstymenko — APPROVED** [review]
Looks good

**Pull request merged**